### PR TITLE
Transpile all code on app browser layer

### PR DESCRIPTION
**Pull request by huozhi**
### What

Include modules from `node_modules` into swc transform, this will help discover server actions from node_modules.

### Why

We didn't transform the code from node_modules in app router browser layer, if you're using server actions in client components from node_modules, it's not going to act properly.

### How

- Remove the `exclude` condition for `node_modules` in app router browser layer
- Change react refresh loader to only apply for userland code (non node_modules) in app router browser layer to avoid broken

There's perf concern that all transform applying to node_modules might cause too much overhead. We're passing down webpack bundling layer to swc loader, to determine when to disable the unnecessary transforms. In this PR we disabled few server specific transform for browser layer.

Closes NEXT-1848
Closes NEXT-1861

### CODE REVIEW

**Comment by ijjk** [comment]
## Failing test suites 
Commit: b18160dc6391f600a2fb72cef7b492f8f8e22859

`pnpm test-dev test/development/acceptance-app/app-hmr-changes.test.ts` 
- Error overlay - RSC build errors > should handle successive HMR changes with errors correctly

<details>
<summary>Expand output</summary>

  ● Error overlay - RSC build errors › should handle successive HMR changes with errors correctly

    expect(received).toContain(expected) // indexOf

    Expected substring: "A few years ago I tweeted"
    Received string:    "<head><meta charset=\"utf-8\"><meta name=\"viewport\" content=\"width=device-width\"><meta name=\"next-head-count\" content=\"2\"><noscript data-n-css=\"\"></noscript><script defer=\"\" nomodule=\"\" src=\"/_next/static/chunks/polyfills.js?ts=1702474444638\"></script><script src=\"/_next/static/chunks/fallback/webpack.js?ts=1702474444638\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/main.js?ts=1702474444638\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/pages/_app.js?ts=1702474444638\" defer=\"\"></script><script src=\"/_next/static/chunks/fallback/pages/_error.js?ts=1702474444638\" defer=\"\"></script><noscript id=\"__next_css__DO_NOT_USE__\"></noscript></head><body style=\"overflow: hidden;\"><div id=\"__next\"></div><script src=\"/_next/static/chunks/fallback/react-refresh.js?ts=1702474444638\"></script><script id=\"__NEXT_DATA__\" type=\"application/json\">{\"props\":{\"pageProps\":{\"statusCode\":500}},\"page\":\"/_error\",\"query\":{},\"buildId\":\"development\",\"isFallback\":false,\"err\":{\"name\":\"ModuleParseError\",\"source\":\"server\",\"message\":\"Module parse failed: Cannot use 'import.meta' outside a module (52:16)\\nFile was processed with these loaders:\\n * ./node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/@next/react-refresh-utils/dist/loader.js\\n * ./node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/build/webpack/loaders/next-flight-client-module-loader.js\\n * ./node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/build/webpack/loaders/next-swc-loader.js\\nYou may need an additional loader to handle the result of these loaders.\\n|                 // still a Refresh Boundary later.\\n|                 // @ts-ignore importMeta is replaced in the loader\\n\\u003e                 import.meta.webpackHot.accept();\\n|                 // This field is set when the previous version of this module was a\\n|                 // Refresh Boundary, letting us know we need to check for invalidation or\",\"stack\":\"ModuleParseError: Module parse failed: Cannot use 'import.meta' outside a module (52:16)\\nFile was processed with these loaders:\\n * ./node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/@next/react-refresh-utils/dist/loader.js\\n * ./node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/build/webpack/loaders/next-flight-client-module-loader.js\\n * ./node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/build/webpack/loaders/next-swc-loader.js\\nYou may need an additional loader to handle the result of these loaders.\\n|                 // still a Refresh Boundary later.\\n|                 // @ts-ignore importMeta is replaced in the loader\\n\\u003e                 import.meta.webpackHot.accept();\\n|                 // This field is set when the previous version of this module was a\\n|                 // Refresh Boundary, letting us know we need to check for invalidation or\\n    at handleParseError (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/webpack/bundle5.js:28:401867)\\n    at /tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/webpack/bundle5.js:28:403614\\n    at processResult (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/webpack/bundle5.js:28:399309)\\n    at /tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/webpack/bundle5.js:28:400370\\n    at /tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:8727\\n    at iterateNormalLoaders (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5565)\\n    at iterateNormalLoaders (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5650)\\n    at /tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5879\\n    at r.callback (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4039)\\n    at Object.ReactRefreshLoader (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/@next/react-refresh-utils/dist/loader.js:14:10)\\n    at LOADER_EXECUTION (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4134)\\n    at runSyncOrAsync (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4145)\\n    at iterateNormalLoaders (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5782)\\n    at iterateNormalLoaders (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5650)\\n    at /tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5879\\n    at r.callback (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4039)\\n    at Object.transformSource (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/build/webpack/loaders/next-flight-client-module-loader.js:34:17)\\n    at LOADER_EXECUTION (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4134)\\n    at runSyncOrAsync (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4145)\\n    at iterateNormalLoaders (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5782)\\n    at /tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5142\\n    at r.callback (/tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4039)\\n    at /tmp/next-install-f6be50af1af88dccd09f7ce3c18cb26c9ff90653ad4e11a5b6a676c80052dffb/node_modules/.pnpm/file+..+next-repo-036d7360dbaaa48fb3d0172511b672473548f87c9b5d8c775debc6d3dfe3afa8+packages+n_zsregpwxswv7uknxpn22yv42jq/node_modules/next/dist/build/webpack/loaders/next-swc-loader.js:167:23\"},\"gip\":true,\"scriptLoader\":[]}</script><div id=\"__next-build-watcher\" style=\"position: fixed; bottom: 10px; right: 20px; width: 0px; height: 0px; z-index: 99999;\"></div><next-route-announcer><p aria-live=\"assertive\" id=\"__next-route-announcer__\" role=\"alert\" style=\"border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; top: 0px; width: 1px; white-space: nowrap; overflow-wrap: normal;\"></p></next-route-announcer><nextjs-portal></nextjs-portal></body>"

      31 |     expect(
      32 |       await session.evaluate('document.documentElement.innerHTML')
    > 33 |     ).toContain('A few years ago I tweeted')
         |       ^
      34 |
      35 |     const pagePath = 'app/(post)/2020/develop-preview-test/page.mdx'
      36 |     const originalPage = await next.readFile(pagePath)

      at Object.toContain (development/acceptance-app/app-hmr-changes.test.ts:33:7)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| buildDuration | 10.7s | 10.8s | ⚠️ +119ms |
| buildDurationCached | 6.2s | 6.1s | N/A |
| nodeModulesSize | 200 MB | 200 MB | N/A |
| nextStartRea..uration (ms) | 425ms | 419ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| 170-HASH.js gzip | 26.8 kB | 26.9 kB | ⚠️ +152 B |
| 199.HASH.js gzip | 181 B | 185 B | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 240 B | 242 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.7 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | N/A |
| Overall change | 72 kB | 72.1 kB | ⚠️ +152 B |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| _app-HASH.js gzip | 195 B | 194 B | N/A |
| _error-HASH.js gzip | 183 B | 182 B | N/A |
| amp-HASH.js gzip | 501 B | 501 B | ✓ |
| css-HASH.js gzip | 321 B | 321 B | ✓ |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | N/A |
| edge-ssr-HASH.js gzip | 255 B | 255 B | ✓ |
| head-HASH.js gzip | 349 B | 350 B | N/A |
| hooks-HASH.js gzip | 368 B | 369 B | N/A |
| image-HASH.js gzip | 4.27 kB | 4.27 kB | N/A |
| index-HASH.js gzip | 255 B | 256 B | N/A |
| link-HASH.js gzip | 2.61 kB | 2.6 kB | N/A |
| routerDirect..HASH.js gzip | 311 B | 310 B | N/A |
| script-HASH.js gzip | 384 B | 384 B | ✓ |
| withRouter-HASH.js gzip | 307 B | 307 B | ✓ |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 1.87 kB | 1.87 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 484 B | ✓ |
| Overall change | 484 B | 484 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| index.html gzip | 529 B | 527 B | N/A |
| link.html gzip | 542 B | 540 B | N/A |
| withRouter.html gzip | 524 B | 523 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| edge-ssr.js gzip | 93.7 kB | 93.7 kB | N/A |
| page.js gzip | 146 kB | 146 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 625 B | 625 B | ✓ |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 37.4 kB | 37.4 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.7 kB | 2.7 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js rsc/transpile-browser | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 168 kB | 168 kB | ✓ |
| app-page-exp..prod.js gzip | 94.1 kB | 94.1 kB | ✓ |
| app-page-tur..prod.js gzip | 94.8 kB | 94.8 kB | ✓ |
| app-page-tur..prod.js gzip | 89.4 kB | 89.4 kB | ✓ |
| app-page.run...dev.js gzip | 138 kB | 138 kB | ✓ |
| app-page.run..prod.js gzip | 88.7 kB | 88.7 kB | ✓ |
| app-route-ex...dev.js gzip | 24 kB | 24 kB | ✓ |
| app-route-ex..prod.js gzip | 16.6 kB | 16.6 kB | ✓ |
| app-route-tu..prod.js gzip | 16.6 kB | 16.6 kB | ✓ |
| app-route-tu..prod.js gzip | 16.2 kB | 16.2 kB | ✓ |
| app-route.ru...dev.js gzip | 23.4 kB | 23.4 kB | ✓ |
| app-route.ru..prod.js gzip | 16.2 kB | 16.2 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.64 kB | 9.64 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| pages.runtim...dev.js gzip | 22.5 kB | 22.5 kB | ✓ |
| pages.runtim..prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| server.runti..prod.js gzip | 49.4 kB | 49.4 kB | ✓ |
| Overall change | 930 kB | 930 kB | ✓ |

</details>

<details>
<summary><strong>Diff details</strong></summary>

<details>
<summary>Diff for <strong>170-HASH.js</strong></summary>

Diff too large to display
</details>
<details>
<summary>Diff for <strong>3f784ff6-HASH.js</strong></summary>

Diff too large to display
</details>

</details>

</details>
Commit: b18160dc6391f600a2fb72cef7b492f8f8e22859

File: test/e2e/app-dir/app-external/app-external.test.ts [hunk 1]
```diff
@@ -251,21 +251,37 @@ createNextDescribe(
...
+            // The prefer esm won't effect turbopack resolving
+            expect(text).toBe('dual-pkg-optout:mjs')
+          } else {
+            expect(text).toBe('dual-pkg-optout:cjs')
+          }
+          return 'success'
+        }, /success/)
+      })
+
+      if (!process.env.TURBOPACK) {
```
Comment (sokra) [inline test/e2e/app-dir/app-external/app-external.test.ts:272]: Avoid that. Just keep it unconditional and use the turbopack-test-manifest to flag it as failed. This way we can track the progress of that.

**Review by shuding — APPROVED** [review]
Thanks!

**Review by feedthejim — APPROVED** [review]

**Comment by joacub** [comment]
This is causing some var export  to be undefined as other is referencing in other issue

**Comment by balazsorban44** [comment]
@joacub could you open/point to an issue about that? :pray:

**Comment by joacub** [comment]
> @joacub could you open/point to an issue about that? 🙏

i already did and someone just close the issue automatically, i haven o time to reproduce entirely, you can do by your self just including a react-pdf in any part of your code.

**Comment by joacub** [comment]
this is the issue:
[
](https://github.com/vercel/next.js/issues/59894)

**Comment by joacub** [comment]
This still causing a lot of issues please some one fix this, is not transforming correctly and removing exports and not picking the right files from packages.json 🥲

**Comment by joacub** [comment]
Nobody going to fix this ?? My got

**Comment by kevva** [comment]
Also seeing some issues with this. Getting the following with a CJS module imported in a ESM project:

> Error: ES Modules may not assign module.exports or exports.*, Use ESM export syntax, instead: …

**Comment by joacub** [comment]
The issue still after the 35 canary version, is there a option i can put to avoid transpiling node modules ?

**Comment by kevva** [comment]
The issues seems to be resolved in `14.0.5-canary.36`.

**Pull request merged**